src/dft/src/architect/ScanArchitectHeuristic.cpp

```diff
@@ -32,21 +32,97 @@
 
 #include "ScanArchitectHeuristic.hh"
 
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/geometry/index/rtree.hpp>
 #include <iostream>
 #include <sstream>
 
 #include "ClockDomain.hh"
 #include "ScanArchitect.hh"
 
+namespace bg = boost::geometry;
+namespace bgi = boost::geometry::index;
+
 namespace dft {
 
 ScanArchitectHeuristic::ScanArchitectHeuristic(
     const ScanArchitectConfig& config,
-    std::unique_ptr<ScanCellsBucket> scan_cells_bucket)
-    : ScanArchitect(config, std::move(scan_cells_bucket))
+    std::unique_ptr<ScanCellsBucket> scan_cells_bucket,
+    utl::Logger* logger)
+    : ScanArchitect(config, std::move(scan_cells_bucket)), logger_(logger)
 {
 }
 
+void ScanArchitectHeuristic::OptimizeScanWirelength(
```

Comment:
```suggestion
void ScanArchitectHeuristic::OptimizeScanWireLength(
```

Reply:
wirelength is normally a single word in EDA contexts, so I think this okay